src/custom/state/orders/updater.ts

```diff
@@ -0,0 +1,176 @@
+import { useEffect, useMemo } from 'react'
+import { useDispatch } from 'react-redux'
+import { useActiveWeb3React } from 'hooks'
+import { useAddPopup, useBlockNumber } from 'state/application/hooks'
+import { AppDispatch } from 'state'
+// import { removeOrder, /* fulfillOrder, */ OrderFromApi } from './actions'
+import { utils } from 'ethers'
+import { Web3Provider } from '@ethersproject/providers'
+import { Log, Filter } from '@ethersproject/abstract-provider'
+import { useLastCheckedBlock } from './hooks'
+import { updateLastCheckedBlock } from './actions'
+// import { PartialOrdersMap } from './reducer'
+
+// example of event watching + decoding without contract
+const transferEventAbi = 'event Transfer(address indexed from, address indexed to, uint amount)'
+const ERC20Interface = new utils.Interface([transferEventAbi])
+
+const TransferEvent = ERC20Interface.getEvent('Transfer')
+
+const TransferEventTopics = ERC20Interface.encodeFilterTopics(TransferEvent, [])
+
+// const decodeTransferEvent = (transferEventLog: Log) => {
+//   return ERC20Interface.decodeEventLog(TransferEvent, transferEventLog.data, transferEventLog.topics)
+// }
+
+type RetryFilter = Filter & { fromBlock: number; toBlock: number }
+
+const constructGetLogsRetry = (provider: Web3Provider) => {
+  // tries fetching logs
+  // if breaks, retries on half the range
+  const getLogsRetry = async ({ fromBlock, toBlock, ...rest }: RetryFilter): Promise<Log[]> => {
+    try {
+      console.log(`EventUpdater::Getting logs fromBlock: ${fromBlock} toBlock ${toBlock}`)
+      const logs = await provider.getLogs({
+        fromBlock,
+        toBlock,
+        ...rest
+      })
+      // console.log('logs', logs)
+      return logs
+    } catch (error) {
+      console.error(`Error getting logs fromBlock: ${fromBlock} toBlock ${toBlock}`, error)
+
+      // expect -- RPC Error: query returned more than 10000 results --
+      // if a different error - rethrow
+      if (!error?.message?.includes('query returned more than')) throw error
+
+      const midBlock = Math.floor((toBlock + fromBlock) / 2)
+
+      const [beforeMidLogs, afterMidLogs] = await Promise.all([
+        getLogsRetry({
+          fromBlock,
+          toBlock: midBlock
+        }),
+        getLogsRetry({
```

Comment:
Woldn't this make this effect?
[1,2] ---> retry---> [1,2] + [3,2]

Reply:
```
from 1 to let's say 10
[1,10] ---> broke
               [1,5] --> success
               [6,10] --> broke
                              [6,8] --> success
                              [9,10] --> success
now we unwind the recursion
             [6,8].concat([9,10])
          [1,5].concat([6,10])
     [1,10] --final
```

Reply:
yes, I understand the happy-bumpy path

I gave a bad example, but I was wondering about asking for just a block case

Ask for [1,1]
Break:
* Ask for [1,1] 
* Ask for [1,1]

now we unwind the recursion
[1,1].concat([1,1])

or im misssing sth?

Reply:
Now you are right here. If we can't halve the `[1,1]` range, then we are out of luck and should throw.
I'll add it.
Though it should never happen, as I doubt 10000 event would even fit in a block

Reply:
On a second thought, let's skip such _log-heavy_ blocks by `returning []`

Reply:
@anxolin , @W3stside you ok with this kind of opt-out solution? 5615d56

Reply:
Yes, is strange case, it could happen if 1 block has lots of events. we saw sth similar when dealing with dxDAO claiming of reputation. Although probably won't ever happen


I think your commit is not right? unless the range is `[from, to)` that I would thin is not. I think getLogs is inclusive --> `[from, to]`:

```js
provider.getLogs({
        fromBlock,
        toBlock,
...
```

I would just not do the second call if `from==to`:
https://github.com/gnosis/gp-v2-swap/commit/5615d56bae5d7cc6ba6ea045f5fb2fe4be4569c1#diff-19e51f3e6a8cdbf2fcffe76fd8ddbebe0adcbf4019205005cdaafdb8c150fc31R63-R66

Instead, I would return `Promise.resolve([])`

Reply:
Now I don't understand you

Yes the range is inclusive.
So if for example we call with `[1,3]` first:
```
[1,3]  --> breaks
     [1,1] --> succeeds
     [2,3] --> breaks
           [2,2] --> succeeds
           [3,3] --> breaks, checks toBlock === fromBlock -->returns []
```
We in agreement so far? Good

For simplicity let's look at a singular breaking call
```ts
getLogsRetry({fromBlock:1, toBlock:1}) {
 try {
  // obviously we try first
  await provider.getLogs({fromBLock, toBlock})
} catch (error) {
//if breaks, and range is [1,1], which can't be divided further
//opt out
if (toBlock === fromBlock) return []

// recursion starts here
```

> I would just not do the second call if from==to

Which second call, we do only one call for every unique range. If you are talking about second call in `Promise.all`, then l see my first example above: it can be `Promise.all([1,1], [2,3])`, different calls, not mutually exclusive.

> Instead, I would return Promise.resolve([])

That's what I do,no? `return []` inside an `async function`

I really don't understand you :man_shrugging:

Reply:
Aha, now I understand you!

My point was that if it fails, with one block. You don't split it and do 2 recursive calls. You just do one. 
Now I understand my mistake, you are not retrying for any error, you are dividing only in the case of too many events. So retrying won't help. 

In this case, it should be an error imo, not an empty array. This will fail silently. And although is more resilient, that wouldn't be my solution if I find too many events in one block. So for now I think is better to throw

Reply:
Well, if you prefer to throw we can throw
But that would throw away (pun intended) all past events, so all orders will be stuck unupdated
Whereas if only some orders would fail to update status, user can ignore them

Still wanna throw?

Reply:
I think so. 10K is enough events. So I prefer not to silence possible issue. If we come to single blocks of that amount of tx, I would say is a nice problem to have, then we can also look on how to handle the cases (maybe by then we are in a WS API model :P )